src-v5/carousel.tsx

```diff
@@ -70,65 +113,74 @@ export const Carousel = (props: CarouselProps): React.ReactElement => {
-      currentSlide,
-      currentSlide - slidesToScroll,
-      count
-    );
-
-    const nextIndex = getNextIndex(to);
-    typeof to === 'number' && props.beforeSlide(slide, nextIndex);
-    !props.disableAnimation && setAnimation(true);
-    setCurrentSlide(to ?? currentSlide);
-    setTimeout(
-      () => {
-        if (!isMounted.current) return;
-        typeof to === 'number' && props.afterSlide(nextIndex);
-        !props.disableAnimation && setAnimation(false);
-      },
-      !props.disableAnimation ? props.speed || 500 : 40
-    ); // if animation is disabled decrease the speed to 40
-  };
+  const moveSlide = useCallback(
+    (to?: number) => {
+      const nextIndex = getNextIndex(to);
+      typeof to === 'number' && beforeSlide(slide, nextIndex);
+
+      !disableAnimation && setAnimationEnabled(true);
 
-  const nextSlide = () => {
-    if (props.wrapAround || currentSlide < count - props.slidesToScroll) {
+      if (typeof to === 'number') {
+        setCurrentSlide(to);
+      }
+
+      setTimeout(
+        () => {
+          if (!isMounted.current) return;
+          typeof to === 'number' && afterSlide(nextIndex);
+          !disableAnimation && setAnimationEnabled(false);
+        },
+        !disableAnimation ? propsSpeed || 500 : 40
+      ); // if animation is disabled decrease the speed to 40
+    },
+    [slide, afterSlide, beforeSlide, disableAnimation, getNextIndex, propsSpeed]
+  );
+
+  const nextSlide = useCallback(() => {
+    if (wrapAround || currentSlide < count - propsSlidesToScroll) {
       const nextPosition = getNextMoveIndex(
-        props.scrollMode,
-        props.wrapAround,
+        scrollMode,
+        wrapAround,
         currentSlide,
         count,
-        props.slidesToScroll
+        propsSlidesToScroll
       );
+
       moveSlide(nextPosition);
-    } else {
-      moveSlide();
     }
-  };
+  }, [
+    count,
+    currentSlide,
+    moveSlide,
+    propsSlidesToScroll,
+    scrollMode,
+    wrapAround
+  ]);
 
-  const prevSlide = () => {
+  const prevSlide = useCallback(() => {
     // boundary
-    if (props.wrapAround || currentSlide > 0) {
+    if (wrapAround || currentSlide > 0) {
       const prevPosition = getPrevMoveIndex(
-        props.scrollMode,
-        props.wrapAround,
+        scrollMode,
+        wrapAround,
         currentSlide,
-        props.slidesToScroll
+        propsSlidesToScroll
       );
+
       moveSlide(prevPosition);
-    } else {
-      moveSlide();
     }
-  };
+  }, [currentSlide, moveSlide, propsSlidesToScroll, scrollMode, wrapAround]);
 
-  useEffect(() => {
-    if (typeof props.slideIndex === 'number' && !props.autoplayReverse) {
-      moveSlide(props.slideIndex);
-    }
-  }, [props.slideIndex]);
+  // Why is this needed?
+  // useEffect(() => {
+  //   if (typeof slideIndex === 'number' && !autoplayReverse) {
+  //     moveSlide(slideIndex);
+  //   }
+  // }, [slideIndex, autoplayReverse, moveSlide]);
 
+  // Why is this needed?
```

Comment:
It would also be helpful to understand this one too! @ValGeorgiev

Reply:
Added comments describing what's is the purpose of this hook